### ISSUE TITLE
Potential optimization idea: Avoid wrapping or boxing

### ISSUE THREAD

**Issue by fzyzcjy** [ethanblake4/dart_eval#64]
> For example, not sure what "wrap" is (not check code yet), but if possible, we should avoid a wrapping, since that generate more objects, i.e. more indirection (slower) and more garbage (gc slower).

in https://github.com/ethanblake4/dart_eval/issues/57#issuecomment-1284853344

**Comment by ethanblake4** [comment]
Boxing is _very_ core to how dart_eval functions (and is inspired by the way Dart itself functions). 

The only reasonable option besides boxing is to tag every object created natively in dart_eval with a concrete runtime type set using an Expando (since this is probably something we'll have to do anyway) and do function lookups on the top type. It would be slower for method calls but there's a good chance it'd be worth it... the problem is bridging though. We'd have to do some sort of global type hash lookup for every single instance that's passed into dart_eval from Dart which would be vastly slower than the current method. Actually, it may be necessary to add this global type hash lookup anyway for return values of generic functions like Future.then, but at least in that case it would only be used sparingly.

Tl;dr the time investment required to remove boxing is probably not worth it, performance might be situationally better but it would also have some severe performance pitfalls.

**Comment by fzyzcjy** [comment]
Sounds reasonable!
